crates/topos-test-sdk/src/integration/mod.rs

```diff
@@ -0,0 +1,154 @@
+use futures::{future::join_all, StreamExt};
+use rand::seq::SliceRandom;
+use std::time::Duration;
+use tokio::time::error::Elapsed;
+use tonic::transport::Uri;
+
+use topos_core::{
+    api::grpc::{
+        shared::v1::checkpoints::TargetCheckpoint,
+        tce::v1::{
+            api_service_client::ApiServiceClient,
+            console_service_client::ConsoleServiceClient,
+            watch_certificates_request::OpenStream,
+            watch_certificates_response::{CertificatePushed, Event},
+            StatusRequest, SubmitCertificateRequest,
+        },
+    },
+    uci::{Certificate, CERTIFICATE_ID_LENGTH, SUBNET_ID_LENGTH},
+};
+use tracing::{debug, info};
+
+pub async fn check_certificate_delivery(
+    timeout_broadcast: u64,
+    peers: Vec<Uri>,
+    timeout: u64,
+) -> Result<Result<(), Vec<String>>, Elapsed> {
+    tokio::time::timeout(Duration::from_secs(timeout), async move {
+        let random_peer: Uri = peers
+            .choose(&mut rand::thread_rng())
+            .ok_or_else(|| {
+                vec![format!(
+                    "Unable to select a random peer from the list: {peers:?}"
+                )]
+            })?
+            .try_into()
+            .map_err(|_| vec![format!("Unable to parse the peer address")])?;
+
+        let pushed_certificate = Certificate::new_with_default_fields(
+            [0u8; CERTIFICATE_ID_LENGTH],
+            [1u8; SUBNET_ID_LENGTH].into(),
+            &[[2u8; SUBNET_ID_LENGTH].into()],
+        )
+        .map_err(|_| vec![format!("Unable to create the certificate")])?;
+
+        let certificate_id = pushed_certificate.id;
+        let mut join_handlers = Vec::new();
+
+        // check that every nodes delivered the certificate
+        for peer in peers {
+            join_handlers.push(tokio::spawn(async move {
+                let peer_string = peer.clone();
+                let mut client = ConsoleServiceClient::connect(peer_string.clone())
+                    .await
+                    .map_err(|_| (peer_string.clone(), "Unable to connect to the api console"))?;
```

Comment:
api console? what is that?

Reply:
@dvdplm I just move this whole chunk of code to a new function, I didn't update it. For name choice probably better for @Freyskeyd to comment